boot/assets.go

```diff
@@ -790,21 +795,24 @@ func observeSuccessfulBootAssetsForBootloader(m *Modeenv, root string, opts *boo
 		return nil, nil
 	}
 
+	byAssetName := make(map[string]string)
+	for trustedAsset, assetName := range trustedAssets {
+		_, alreadyHere := byAssetName[assetName]
+		if alreadyHere {
+			logger.Noticef("bootloader %s has several asset of the same name %s", whichBootloader, assetName)
+		}
+		byAssetName[assetName] = trustedAsset
+	}
+
 	cache := newTrustedAssetsCache(dirs.SnapBootAssetsDir)
-	for _, trustedAsset := range trustedAssets {
-		assetName := filepath.Base(trustedAsset)
 
-		// find the hash of the file on disk
+	for assetName, trustedAsset := range byAssetName {
 		assetHash, err := cache.fileHash(filepath.Join(root, trustedAsset))
-		if err != nil && !os.IsNotExist(err) {
-			return nil, fmt.Errorf("cannot calculate the digest of existing trusted asset: %v", err)
-		}
-		if assetHash == "" {
-			// no trusted asset on disk, but we booted nonetheless,
-			// at least log something
```

Comment:
we probably want to keep this comment before the log below?

Reply:
Not all trusted assets will be present now. So I am thinking we should remove the notice. But the logic is still fine. If the asset is not in the cache, then we should ignore it.